Consul/Agent.cs

```diff
@@ -776,6 +781,7 @@ public void Dispose()
             {
                 _streamreader.Dispose();
                 _stream.Dispose();
+                GC.SuppressFinalize(this);
```

Comment:
It was a lint suggestion and I usually use it on my `Dispose` methods as well to prevent unnecessary GC calls.

Reply:
What kind of linter you use? I would like to try it myself, as I don't see why this method would ever require to call the `GC.SuppressFinalize` as it doesn't even has the finalizer.

Reply:
I am using EditorConfig. It's using the ruleset that came with the repository. See [here](https://github.com/G-Research/consuldotnet/blob/master/.editorconfig)

Reply:
I've never investigated this before but I think the main linting engine is the [C# for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp), which uses the ruleset in EditorConfig.

Reply:
VS Code with that extension doesn't show this lint suggestion to me. 
Anyway, the CA1816 is confusing to people, see https://github.com/dotnet/roslyn-analyzers/issues/4801. 

Apparently, if the class is unsealed, it should implement the dispose pattern like this -> https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern#basic-dispose-pattern. We cannot mark the `LogStream` class sealed now because it would be introducing incompatibility with previous versions. So in our case, we need to implement the "Basic Dispose Pattern" like this:
```
            public void Dispose()
            {
                Dispose(true);
                GC.SuppressFinalize(this);
            }

            protected virtual void Dispose(bool disposing)
            {
                if (disposing)
                {
                    // Dispose managed resources.
                    _streamreader.Dispose();
                    _stream.Dispose();
                }
            }
```

Reply:
Okay, do we implement the way you suggested, or do we revert and I can add a EditorConfig rule to block that lint?